### PR TITLE
[yugabyte/yugabyte-db#19196] Add test to ensure no duplicate snapshot

### PR DESCRIPTION
**Pull request by vaibhav-yb**
## Problem

The problem was identified as a service side problem, the RCA for which has been added in the ticket https://github.com/yugabyte/yugabyte-db/issues/19196

## Solution

The solution was also added as a commit on service side for the ticket https://github.com/yugabyte/yugabyte-db/issues/19348

### Test plan

The current PR only adds a test to the connector UTs to ensure we do not regress in the future. Can be run using:

```sh
mvn clean test -Dtest=YugabyteDBSnapshotTest#snapshotShouldNotBeAffectedByDroppingUnrelatedTables
```

### CODE REVIEW

**Review by dr0pdb — APPROVED** [review]
Looks ok to me at a high level but I don't have full context on connector. Do wait for Suranjan's review.

File: src/main/java/io/debezium/connector/yugabytedb/YugabyteDBStreamingChangeEventSource.java [hunk 1]
```diff
@@ -990,6 +992,14 @@ protected void handleTabletSplit(String splitTabletId,
 
         String tableId = entryToBeDeleted.getKey();
 
+        if (TEST_WAIT_BEFORE_GETTING_CHILDREN) {
+            final int secondsToWait = 60;
+            LOGGER.info("[TEST ONLY] Waiting for {} seconds before getting children for tablet {}",
+                        secondsToWait, splitTabletId);
+            final Metronome waitMetronome = Metronome.parker(Duration.ofSeconds(secondsToWait), Clock.SYSTEM);
+            waitMetronome.pause();
```
[the remark this answers is not in the document]
Comment (vaibhav-yb) [reply src/main/java/io/debezium/connector/yugabytedb/YugabyteDBStreamingChangeEventSource.java:1000]: Yes, waiting before getting children is meant for the service. The behaviour we were seeing was that service ended up deleting the parent tablet and we wanted to replicate that.

File: src/test/java/io/debezium/connector/yugabytedb/YugabyteDBSnapshotTest.java [hunk 2]
```diff
@@ -629,6 +630,106 @@ public void shouldContinueStreamingInNeverAfterSnapshotCompleteInInitialOnly(boo
...
+
+        // Split the tablet. There is just one tablet so it is safe to assume that the iterator will
+        // return just the desired tablet.
+        ybClient.splitTablet(tablets.iterator().next());
+
+        // Wait till there are 2 tablets for the table.
+        TestHelper.waitForTablets(ybClient, table, 2);
+
+        LOGGER.info("Dummy wait to see the logs in connector");
+        TestHelper.waitFor(Duration.ofSeconds(120));
```
[the remark this answers is not in the document]
Comment (vaibhav-yb) [reply src/test/java/io/debezium/connector/yugabytedb/YugabyteDBSnapshotTest.java:699]: I will remove this.
